frontend/packages/dev-console/src/components/health-checks/create-health-checks-probe-utils.ts

```diff
@@ -50,14 +50,25 @@ export const getRequestType = (data: HealthCheckProbeData) => {
 export const convertResourceDataToFormData = (
   resourceData: HealthCheckProbeData,
 ): HealthCheckFormProbeData => {
+  if (!resourceData) return null;
   return {
     ...resourceData,
     requestType: getRequestType(resourceData),
-    failureThreshold: resourceData.failureThreshold.toString(),
-    successThreshold: resourceData.successThreshold.toString(),
-    initialDelaySeconds: resourceData.initialDelaySeconds?.toString(),
-    periodSeconds: resourceData.periodSeconds.toString(),
-    timeoutSeconds: resourceData.timeoutSeconds.toString(),
+    ...(resourceData.hasOwnProperty('failureThreshold') && {
+      failureThreshold: resourceData.failureThreshold.toString(),
+    }),
+    ...(resourceData.hasOwnProperty('successThreshold') && {
+      successThreshold: resourceData.successThreshold.toString(),
+    }),
+    ...(resourceData.hasOwnProperty('initialDelaySeconds') && {
+      initialDelaySeconds: resourceData.initialDelaySeconds.toString(),
+    }),
+    ...(resourceData.hasOwnProperty('periodSeconds') && {
+      periodSeconds: resourceData.periodSeconds.toString(),
+    }),
+    ...(resourceData.hasOwnProperty('timeoutSeconds') && {
+      timeoutSeconds: resourceData.timeoutSeconds.toString(),
+    }),
```

Comment:
@vikram-raj @invincibleJai IMHO: we have two types to convert between resource data and form data. I would recommend to not mark form data as optional or return null here (line 53).

This could result in general errors when the component tries to render

```html
<input value={form....successThreshold} />
```

This could result in two errors (both doesn't happen at the moment. its a general)

1. We are sure that the sub-tree of a form is well defined
2. Form fields should not be undefined. Switching between undefined and string values could result in uncontroller/controlled react issues. But this is maybe not necessary with PF input fields.

Instead I would recommend to set the values and add a fallback to `""`. So that we now that the form data are always defined. 

```tsx
successThreshold: resourceData.successThreshold?.toString() || '',
```

Wdyt?

Reply:
makes sense to me to provide defaults here